pytest_tests/testsuites/acl/test_acl.py

```diff
@@ -178,3 +180,149 @@ def test_basic_acl_readonly(self, wallets, client_shell, read_only_container, fi
+        """
+        endpoint = self.cluster.default_rpc_endpoint
+
+        user_wallet = wallets.get_wallet()
+        ir_wallet = wallets.get_ir_wallet()
+        storage_wallet = wallets.get_storage_wallet()
+
+        cid = public_container
+
+        with allure.step("Add test objects to container"):
+            owner_object_oid = put_object_to_random_node(
+                user_wallet.wallet_path, file_path, cid, shell=self.shell, cluster=self.cluster
+            )
+
+        with allure.step("Check IR and STORAGE rules compliance"):
+            assert not can_put_object(
+                ir_wallet.wallet_path,
+                cid,
+                file_path,
+                shell=self.shell,
+                cluster=self.cluster,
+                wallet_config=ir_wallet.config_path,
+            )
+            assert can_put_object(
+                storage_wallet.wallet_path,
+                cid,
+                file_path,
+                shell=self.shell,
+                cluster=self.cluster,
+                wallet_config=storage_wallet.config_path,
+            )
+
+            assert can_get_object(
+                ir_wallet.wallet_path,
+                cid,
+                owner_object_oid,
+                file_path,
+                shell=self.shell,
+                cluster=self.cluster,
+                wallet_config=ir_wallet.config_path,
+            )
+            assert can_get_object(
+                storage_wallet.wallet_path,
+                cid,
+                owner_object_oid,
+                file_path,
+                shell=self.shell,
+                cluster=self.cluster,
+                wallet_config=storage_wallet.config_path,
+            )
+
+            assert can_get_head_object(
+                ir_wallet.wallet_path,
+                cid,
+                owner_object_oid,
+                shell=self.shell,
+                endpoint=endpoint,
+                wallet_config=ir_wallet.config_path,
+            )
+            assert can_get_head_object(
+                storage_wallet.wallet_path,
+                cid,
+                owner_object_oid,
+                shell=self.shell,
+                endpoint=endpoint,
+                wallet_config=storage_wallet.config_path,
+            )
+
+            assert can_search_object(
+                ir_wallet.wallet_path,
+                cid,
+                shell=self.shell,
+                endpoint=endpoint,
+                oid=owner_object_oid,
+                wallet_config=ir_wallet.config_path,
+            )
+            assert can_search_object(
+                storage_wallet.wallet_path,
+                cid,
+                shell=self.shell,
+                endpoint=endpoint,
+                oid=owner_object_oid,
+                wallet_config=storage_wallet.config_path,
+            )
+
+            with pytest.raises(AssertionError):
+                assert can_get_range_of_object(
```

Comment:
AssertionError is expected to be raised from the assert here? Or from somewhere internally? If from this assert, maybe better to use just ```assert no can_get...```

Reply:
Also it is not clear from the test why some operations are allowed and others not. It is better to add some related comment

Reply:
> AssertionError is expected to be raised from the assert here? Or from somewhere internally? If from this assert, maybe better to use just `assert not can_get...`

Fixed

Reply:
> Also it is not clear from the test why some operations are allowed and others not. It is better to add some related comment

Fixed